src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3270,6 +3277,43 @@ internal bool CanEmitSpecialType(SpecialType type)
             return (diagnostic == null) || (diagnostic.Severity != DiagnosticSeverity.Error);
         }
 
+        internal bool EmitNullablePublicOnly
+        {
+            get
+            {
+                if (!_lazyEmitNullablePublicOnly.HasValue())
+                {
+                    bool value = SyntaxTrees.FirstOrDefault()?.Options?.Features?.ContainsKey("nullablePublicOnly") == true;
+                    _lazyEmitNullablePublicOnly = value.ToThreeState();
+                }
+                return _lazyEmitNullablePublicOnly.Value();
+            }
+        }
+
+        internal bool ShouldEmitNullableAttributes(Symbol symbol)
+        {
+            Debug.Assert(!(symbol is null));
+            Debug.Assert(symbol.IsDefinition);
+
+            var sourceAssembly = SourceAssembly;
+            if (symbol.ContainingAssembly != sourceAssembly)
```

Comment:
>if (symbol.ContainingAssembly != sourceAssembly) [](start = 12, length = 48)

Is this check for just in case, or do we have a scenario when it is significant? Consider switching to SourceModule comparison instead.

Reply:
We hit this in the EE.

---
In reply to: [293924012](https://github.com/dotnet/roslyn/pull/36398#discussion_r293924012) [](ancestors = 293924012)

Reply:
> We hit this in the EE.

The comparing the module is the right check instead, I think.

---
In reply to: [293924981](https://github.com/dotnet/roslyn/pull/36398#discussion_r293924981) [](ancestors = 293924981,293924012)